### PR TITLE
SoundWire: recheck unattached status

### PR DESCRIPTION
**Pull request by plbossart**
New attempt to solve issues with two amplifiers on the same link. The latest CI results indicate that the UNATTACHED even is a false positive, this should be doubled-checked with the PING frame status.

### CODE REVIEW

File: drivers/soundwire/bus.c [hunk 1]
```diff
@@ -1799,6 +1799,25 @@ int sdw_handle_slave_status(struct sdw_bus *bus,
 		    slave->status != SDW_SLAVE_UNATTACHED) {
 			dev_warn(&slave->dev, "Slave %d state check1: UNATTACHED, status was %d\n",
 				 i, slave->status);
+
+			if (bus->recheck_unattached && bus->ops->read_ping_status) {
+				u32 ping_status;
+
+				mutex_lock(&bus->msg_lock);
+
+				ping_status = bus->ops->read_ping_status(bus);
```
Comment (bardliao) [inline drivers/soundwire/bus.c:1808]: I am thinking if we will get the same result when we read ping status and from `cdns_update_slave_status`. If so, we may still see the issue if the ping status is still UNATTACHED (and will become ATTACHED later) when we read it here.

Comment (plbossart) [reply drivers/soundwire/bus.c:1808]: @bardliao if the ping status is UNATTACHED, then the device should re-appear as Device0 and the existing logic for the enumeration should work.

What doesn't work is when a device is reported as UNATTACHED, but that's a false positive. In that case, the device will NEVER report as Device0 and will never be re-enumerated again. The logic will wait for something that will not happen and times out.

**Review by RanderWang — APPROVED** [review]
LGTM. If we could reproduce it and use analyzer to get more info, it should be a interesting topic

**Comment by plbossart** [comment]
@bardliao any objections to this patch, i've lost track of your feedback.

**Comment by bardliao** [comment]
> @bardliao any objections to this patch, i've lost track of your feedback.

No objections, I just feel the PR may not solve the issue completely. IMHO, that is more like a timing issue.

**Review by bardliao — APPROVED** [review]

**Comment by plbossart** [comment]
> > @bardliao any objections to this patch, i've lost track of your feedback.
> 
> No objections, I just feel the PR may not solve the issue completely. IMHO, that is more like a timing issue.

I don't think it's a timing issue because we have a positive sighting of an UNATTACHED event. That cannot be created by a timing issue of the software side, it's the Cadence IP that detected the device was no longer attached.
let's merge it and see what happens in CI tests. If we don't see it after a month we are probably in good shape.

**Pull request merged**